### DISCUSSION TITLE
[@types/tap] Missing `t.on` (`t.on('end')`)

### DISCUSSION THREAD

**Discussion by inga-lovinde** [DefinitelyTyped/DefinitelyTyped#64384] [Issues with a @types package]
According to node-tap [documentation](https://node-tap.org/docs/api/test-lifecycle-events/):

> `t.on('end')`
> The `end` event fires when the test is completely finished, and all of its teardown functions have completed.
> This is just a normal `EventEmitter` event, so it doesn't support any sort of async actions.

So this is a valid code:

```
import tap from 'tap';

tap.on('end', () => console.log('suite is done'));

tap.test('test name', async t => {
    t.on('end', () => console.log('test is done'));
});
```

It actually executes and logs expected messages to console.

However, neither `tap` nor `t` are declared as having an `on` field in `@types/tap`.

[a comment by an automated account is not shown]

**Reply by zkldi** [reply]
Hi! This is on my radar. I'll fix this tonight. Thank you.
